File: slack-reader/src/main/java/dev/number6/slackreader/port/SlackPort.java
package dev.number6.slackreader.port;

import dev.number6.slackreader.model.Channel;
import dev.number6.slackreader.model.Message;
import com.amazonaws.services.lambda.runtime.LambdaLogger;

import java.time.LocalDate;
import java.util.Collection;

public interface SlackPort {
    Collection<Channel> getChannelList(LambdaLogger logger);

    Collection<Message> getMessagesForChannelOnDate(Channel c, LocalDate date, LambdaLogger logger);
}

File: slack-reader/src/main/java/dev/number6/slackreader/port/SlackPort.kt
package dev.number6.slackreader.port

import com.amazonaws.services.lambda.runtime.LambdaLogger

import dev.number6.slackreader.model.Channel
import dev.number6.slackreader.model.Message
import java.time.LocalDate

interface SlackPort {
    open fun getChannelList(logger: LambdaLogger): Collection<Channel>
    open fun getMessagesForChannelOnDate(c: Channel, date: LocalDate, logger: LambdaLogger): Collection<Message>
}